### DISCUSSION TITLE
Some questions for choosing evolu

### DISCUSSION THREAD

**Discussion by freeseamew** [evoluhq/evolu#470] [Q&A]
I am currently developing an app that requires offline first functionality.

Then, I happened to find evolu and I thought it fit very well with the concept of the app I am developing.

So I would like to ask a few questions.

1. First of all, I found out that many parts of evolu will be upgraded soon. However, I read an article in X that said they are trying to maintain the existing API as much as possible. I would like to ask if I can start development by referring to the current API with reference to this part.
2. I am currently using svelte as the frontend of the app I am trying to create. According to the documentation, it seems that @evolu/common-web can be developed in an environment other than react. I would like to hear your opinion on this part as well.

I would like to hear your opinion on these parts.

Anyway, evolu seems to be a project that I am very interested in. I hope the update goes smoothly.

**Comment by steida** [comment]
@freeseamew Hi.

1) "many *internal* parts", the public API will be the same. There is no reason to change it. If you want to run your own Evolu Server (a Relay in the new version), you will switch to a new library with the different `syncUrl` value in config.

2) look at Evolu React binding; it's a thin layer. I think Svelte binding (adapter) should be easy. Feel free to ask at https://discord.com/channels/1072319124337856613/1072319124337856616

As for the sync update, it's a culmination of many months of work, and yet, it goes smoothly because I already solved and implemented the most challenging parts. It was developed with TDD outside of Evolu, and I'm incorporating it now. It's also a chance to solve long-term pains like dependency on ProtoBuf. I can't estimate correctly, but I'm working on it every day.

**Reply by steida** [reply]
There is also automatic migration, which is relatively simple, and for data on Evolu Server, we can't migrate them, of course, because they are encrypted, but if you connect Evolu to a new Relay, it will sync everything there.
